src/components/comment/CommentView.jsx

```diff
@@ -1,22 +1,50 @@
-import React from "react";
+import React, {useState} from "react";
 import TimeAgo from "javascript-time-ago";
 import en from "javascript-time-ago/locale/en";
 import Constants from "../../constants/Constants";
 
+const UNKNOWN_AUTHOR = "Unknown author";
+
 const CommentView = (props) => {
+    TimeAgo.addLocale(en);
+    const time = new TimeAgo('en-US');
+
+    const [show, setShow] = useState(false);
+
     const parseJsonComment = (value, constant) => {
         const jsonComment = JSON.parse(JSON.stringify(value));
         return jsonComment[constant];
     }
 
-    TimeAgo.addLocale(en);
-    const time = new TimeAgo('en-US');
-    const timeAgo = time.format(parseJsonComment(props.timestamp, [Constants.HAS_TIMESTAMP]));
+    const getAuthor = () => {
+        if (props.author) {
+            return parseJsonComment(props.author, [Constants.HAS_AUTHOR]);
+        } else return UNKNOWN_AUTHOR;
+    }
+
+    const getSplitAuthor = () => {
+        let commentAuthor = {};
+        if (props.author) {
+            commentAuthor = parseJsonComment(props.author, [Constants.HAS_AUTHOR]);
+            return "..." + commentAuthor.split('/people/').pop();
```

Comment:
This is not correct as you are assuming that url will contain string people. The url can be any url you can imagine.

Take only localName i.e. last fragment of url. Fragments are separated by "/" or "#".